src/idp/mod.rs

```diff
@@ -61,6 +61,13 @@ impl IdentityProvider {
         Ok(IdentityProvider { private_key })
     }
 
+    pub fn from_private_key_pem(der_bytes: &[u8]) -> Result<Self, Error> {
```

Comment:
nit: rename to pem_bytes to avoid confusion

```suggestion
    pub fn from_private_key_pem(pem_bytes: &[u8]) -> Result<Self, Error> {
```